core/query/src/main/java/datawave/core/query/logic/CheckpointableQueryLogic.java

```diff
@@ -0,0 +1,60 @@
+package datawave.core.query.logic;
+
+import java.util.List;
+
+import org.apache.accumulo.core.client.AccumuloClient;
+
+import datawave.core.query.configuration.GenericQueryConfiguration;
+
+public interface CheckpointableQueryLogic {
+
+    /**
+     * This will allow us to check if a query logic is actually checkpointable. Even if the query logic supports it, the caller may have to tell the query logic
+     * that it is going to be checkpointed.
+     *
+     * @return true if checkpointable
+     */
+    boolean isCheckpointable();
```

Comment:
@ivakegg this has always struck me as a little odd.  If the query logic implements CheckpointableQueryLogic, isn't that sign enough that the query logic is checkpointable?  Are you using this as more of an on/off switch?

Reply:
I agree. The only use I can see for this is if you wanted to do A/B testing with checkpointable logic disabled.

Reply:
I think I could get on board for renaming this to something along the lines of 'enableCheckpoints' or something like that.  This is less about whether the query logic is checkpointable, and more about whether we want to use it in a checkpointable fashion.

Reply:
So the setCheckpointable method below would be the method to change to enableCheckpoints if you wish.  Its the CreateTask/NextTask classes used by the query executor service that needs to know whether the query can be checkpointed of not.  If not, then in the create task all of the results are pulled at that point.  If is it then other executor service instances can handle pulling results as needed.  This mechanism allows us to migrate query logics into the query microservices framework but still support those query logics that have not been migrated.

Reply:
I would prefer enableCheckpoints or checkpointsEnabled, but that's not a hill I'm willing to die on.